creator-node/src/URSMService.js

```diff
@@ -0,0 +1,234 @@
+   *    b. Error if all available nodes contacted without 3 successful signatures
+   *  5. Submit registration transaction to URSM with signatures
+   */
+  async init () {
+    this.logInfo('STARTING INIT')
+
+    /**
+     * 1. Fetch node record from L1 ServiceProviderFactory for spID
+     */
+    const spRecordFromSPFactory = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceEndpointInfo(
+      'content-node',
+      this.spID
+    )
+    let {
+      owner: ownerWalletFromSPFactory,
+      delegateOwnerWallet: delegateOwnerWalletFromSPFactory,
+      endpoint: endpointFromSPFactory
+    } = spRecordFromSPFactory
+    delegateOwnerWalletFromSPFactory = delegateOwnerWalletFromSPFactory.toLowerCase()
+
+
+    /**
+     * 1-a. Short-circuit if no L1 record found
+     */
+    if (
+      LibsUtils.isZeroAddress(ownerWalletFromSPFactory)
+      || LibsUtils.isZeroAddress(delegateOwnerWalletFromSPFactory)
+      || !endpointFromSPFactory
+    ) {
+      throw new Error('L1recordbad')
+    }
+
+    /**
+     * 2. Fetch node record from L2 UserReplicaSetManager for spID
+     */
+    const delegateOwnerWalletFromURSM = (
+      (await this.audiusLibs.contracts.UserReplicaSetManagerClient.getContentNodeWallets(this.spID))
+      .delegateOwnerWallet
+    ).toLowerCase()
+
+    /**
+     * 2-a. Short-circuit if L2 record for node already matches L1 record (i.e. delegateOwnerWallets match)
+     */
+    if (delegateOwnerWalletFromSPFactory === delegateOwnerWalletFromURSM) {
+      throw new Error('L2recordmatch')
+    }
+
+    /**
+     * 3. Fetch list of all nodes registered on URSM, in order to submit requests for proposal
+     *  a. Randomize list to minimize bias
+     *  b. Remove duplicates by owner_wallet key due to on-chain uniqueness constraint
+     */
+    let URSMContentNodes = await this.audiusLibs.discoveryProvider.getURSMContentNodes()
+    
+    // TODO - setting list to single value for testing purposes
+    const tmp = URSMContentNodes[0]
+    this.logInfo(`tmp: ${JSON.stringify(tmp, null, 2)}`)
+
+    URSMContentNodes = URSMContentNodes.filter(node => node.endpoint)
+    URSMContentNodes = _.shuffle(URSMContentNodes)
+    URSMContentNodes = Object.values(_.keyBy(URSMContentNodes, 'owner_wallet'))
+
+    /**
+     * TODO document
+     */
+    // let numRemainingSignatures = 1
+    let numRemainingSignatures = 3
+    // let availableNodes = [tmp]
+    let availableNodes = URSMContentNodes
+    let receivedSignatures = []
+
+    while (numRemainingSignatures > 0) {
+      if (availableNodes.length < numRemainingSignatures) {
+        // TODO what is appropriate way to error -> kill proc, set flag to flip health check?
+        throw new Error('Failure to receive required attestation signatures for URSM Registration')
+      }
+
+      let nodesToAttempt = availableNodes.slice(0, numRemainingSignatures)
+      availableNodes = availableNodes.slice(numRemainingSignatures)
```

Comment:
if 1/3 requests in the first iteration fails, do we move to indexes 4-6 or exclude the failed node only?

Reply:
seems like we'd want to exclude only the failures instead of requesting another 3

Reply:
discussed offline, correct as is